src/controllers/user-controller.js

```diff
@@ -15,3 +17,19 @@ export const findAllUsers = (obj, args, { models }) => {
 export const findUserById = (_, { id }, { models }) => {
   return models.user.findByPk(id)
 }
+
+export const signIn = async (_, { data }, { models }) => {
+  const user =  models.user.findUserByUsername( data.username )
+  if (user){
+    const secret = process.env.JWT_SECRET;
+    const token = jwt.sign({ sub: user.id }, secret);
+    return (
+      user,
+      token
+    )
+  }
+  else {
+    throw new Error ('El usuario no existe')
```

Comment:
Don't give a clue to a potential hacker. The authentication error should be "Invalid username or password", so that the "hacker" doesn't know if what's wrong is the username or the password. Otherwise, you're giving him information to hack your system.

Also, you're not validating the password.